application/modules/core/helpers/ContentBlockHelper.php

```diff
@@ -0,0 +1,197 @@
+     * @return {string}            Result string ready for replacing
+     */
+    public static function compileChunk($chunk, $arguments) {
+        $matches = [];
+        $formatParams = null;
+        preg_match_all('%['.static::$prefix.']{2}([\+\*\%])([^\]\[]+)['.static::$suffix.']{2}%ui', $chunk, $matches);
+        foreach ($matches[2] as $k => $rawParam) {
+            $token = $matches[1][$k];
+            if (false === mb_strpos($rawParam, ':')) {
+                $paramName = $rawParam; 
+            } else {
+                list($paramName, $formatParams) = explode(':', $rawParam);
+            }
+            switch ($token) {
+                case '+':
+                    if (array_key_exists($paramName, $arguments)) {
+                        $replacement = static::applyFormatter($arguments[$paramName], $formatParams);
+                        $chunk = str_replace($matches[0][$k], $replacement, $chunk);
+                    } else if(array_key_exists($paramName.'-default', $arguments)) {
+                        $replacement = static::applyFormatter($arguments[$paramName.'-default'], $formatParams);
+                        $chunk = str_replace($matches[0][$k], $replacement, $chunk);
+                    } else {
+                        $chunk = str_replace($matches[0][$k], '', $chunk);
+                    }
+                    break;
+                default:
+                    $chunk = str_replace($matches[0][$k], '', $chunk);
+            }
+        }
+        return $chunk;
+    }
+
+    /**
+     * Find formatter declarations in chunk placeholders. if find trying to apply
+     * yii\i18n\Formatter formats see yii\i18n\Formatter for details
+     * @param {string} $rawParam single placeholder declaration from chunk
+     * @param $rawFormat {string}
+     * @return array
+     */
+    private static function applyFormatter($value, $rawFormat)
+    {
+        if (null === $rawFormat) {
+            return $value;
+        }
+        $params = explode(',', $rawFormat);
+        $method = array_shift($params);
+        $formattedValue = call_user_func_array([Yii::$app->formatter, $method], [$value, $params]);
+        return $formattedValue;
+    }
+
+    /**
+     * Fetches single chunk by key from static var
+     * if is no there - get it from db and push to static array
+     * @param $key {string} Chunk key field
+     * @return {string} Chunk value field
+     */
+    public static function fetchChunkByKey($key) {
+        if (!array_key_exists($key, static::$chunksByKey)) {
+            $chunkCacheKey = 'chunkCaheKey'.$key.ContentBlock::className();
+            static::$chunksByKey[$key] = Yii::$app->cache->get($chunkCacheKey);
+            if (false === static::$chunksByKey[$key]) {
+                $chunk = ContentBlock::find()
+                    ->where(['key' => $key])
+                    ->asArray()
+                    ->one();
```

Comment:
Try this http://www.yiiframework.com/doc-2.0/guide-caching-data.html#query-caching looks much better